bgpd/bgp_table.h

```diff
@@ -166,6 +156,8 @@ static inline struct bgp_dest *bgp_dest_parent_nolock(struct bgp_dest *dest)
 {
 	struct route_node *rn = bgp_dest_to_rnode(dest)->parent;
 
+	while (rn && !rn->info)
```

Comment:
Just to understand what's going on: Why this loop is needed everywhere on the helper functions?

Reply:
Because now we create bgp_dest based on the need, not all rn->info have a corresponding bgp_dest. If we want to get the dest, we need to skip those nodes where rn->info is empty.